bot/commands.js

```diff
@@ -0,0 +1,236 @@
+            }
+
+            break;
+
+        case (command === userPrefix + "playingnow"): // $playingnow
+
+            playingnow(msg)
+
+            break;
+
+        case (command === userPrefix + 'popcorn'): // $popcorn
+
+            // $popcorn TODO: repalce this with any gif with command $gif 'string'
+
+            popcorn(msg);
+
+            break;
+
+        case (command === userPrefix + 'stats'): // $stats
+
+            // For stats commands the server needs a moment to respond to the request
+            // Here we initially display some placeholder text and when the server provides
+            // its response we replace it with the actual data.
+
+            // TODO: When a user issues the command with no parameter, reference there own BattleTag if in database.
+
+            switch (parameters.length)
+            {
+                case 0:
+
+                    help(msg, command, "Shows you a user's Overwatch quick play statistics.", "[BattleTag]", "[BattleTag] The Battle.net Tag of the user you would like to lookup.");
+
+                    break;
+
+                case 1:
+
+                    msg.reply("Working on your request.")
+                        .then(message =>
+                        {
+                            const msgID = message.id;
+                            stats(msg, msgID);
+                        })
+                        .catch(console.error);
+
+                    break;
+
+                default:
+
+                    help(msg, command, "Shows you a user's Overwatch quick play statistics.", "[BattleTag]", "[BattleTag] The Battle.net Tag of the user you would like to lookup.");
+            }
+
+            break;
+
+        case (command === userPrefix + 'test'): // $test
+
+            // Use this command to execute some code via the "test" command on the test server
+
+            if (bot.user.tag == testbotjoe26Tag)
+            {
+                test(bot, msg);
+            }
+
+            break;
+
+        default: // Command not found
+
+            msg.channel.send('', {
+                embed: {
+                    color: 0xFF0000,
+                    fields: [
+                        {
+                            name: command + ' not found',
+                            value: '```yaml\n' + `Type $commands to display a list of available commands` + '```'
+                        }
+                    ]
+                }
+            });
+    }
+}
+
+module.exports =
+{
+    process: function (bot, msg)
+    {
+        if (msg.content.startsWith(userPrefix) || msg.content.startsWith(adminPrefix))
+        {
+            var data;
+
+            var command;
+
+            var parameters;
+
+            data = msg.content.match(/(['"])((?:\\\1|.)+?)\1|([^\s"']+)/g);
+
+            command = data[0];
+
+            data.splice(0, 1);
+
+            parameters = data;
```

Comment:
It doesn't look like parameters is actually ever used. Can this be removed?

Reply:
Removed both the local var parameters and the unused line parameters = data as this was used during testing and is no longer needed.

Reply:
Parameter has been updated to be used in the function call.